### PR TITLE
Generic Run Analogues

### PR DESCRIPTION
**Pull request by GrigoriiSolnyshkin**
With this pull request functions analogous to Kotlin `run` successfully compile to Viper if they're inline.
For instance, call to
```inline fun <T, R> T.copiedRun(block: T.() -> R): R = block()```
will be correctly inlined.

Support for real stdlib `run` and other notable extension functions (`also`, `let` etc.) as well as the cases with multiple possible receivers will be implemented in future PRs.

### CODE REVIEW

File: plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConversionVisitor.kt [hunk 1]
```diff
@@ -284,7 +301,7 @@ object StmtConversionVisitor : FirVisitor<ExpEmbedding, StmtConversionContext>()
         return when (typeOperatorCall.operation) {
             FirOperation.IS -> Is(argument, conversionType)
             FirOperation.NOT_IS -> Not(Is(argument, conversionType))
-            FirOperation.AS -> Cast(argument, conversionType).withAccessAndProvenInvariants()
+            FirOperation.AS -> argument.withNewTypeAccessAndProvenInvariants(conversionType)
```
Comment (jesyspa) [inline plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConversionVisitor.kt:304]: Is this a fix of an earlier bug?
I understand it can be a bit of a pain, but I would encourage splitting the small unrelated changes into a separate PR so that that can be accepted quickly.

Comment (GrigoriiSolnyshkin) [reply plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConversionVisitor.kt:304]: OK, I think it was just some optimization that would rarely work anyway.
Okay, I'll trying doing so.

File: plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ParameterResolver.kt
[the remark this answers is not in the document]
Comment (GrigoriiSolnyshkin) [reply plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ParameterResolver.kt:1]: I did it to avoid too many `SpecialNames.THIS` calls. Besides, the implementation for non-inline parameters is a little bit different.
I think I will need such function but with parameters in the upcoming PRs to find a proper receiver.
It turns out that explicit `this` in Kotlin can only refer to the innermost scope (with receiver argument), but implicit "this" may come from anywhere. With parameters the situation is different.

**Comment by jesyspa** [comment]
The thing that concerns me here is that we are collapsing two different notions (dispatch receiver and extension receiver) into one.  Maybe if we're working on this anyway, it would be best if we supported both?  Or is there some technical reason why this would be much harder?

**Comment by GrigoriiSolnyshkin** [comment]
> Or is there some technical reason why this would be much harder?

@jesyspa, well, in this case we would need to distinguish possible implicit receivers already.

**Comment by jesyspa** [comment]
Let's have a call about this on Monday, I'm not entirely against doing it this way but it seems risky.

**Review by jesyspa — APPROVED** [review]
